src/common/cron.cc

```diff
@@ -58,9 +63,21 @@ bool Cron::IsTimeMatch(const tm *tm) {
     return false;
   }
   for (const auto &st : schedulers_) {
-    if ((st.minute == -1 || tm->tm_min == st.minute) && (st.hour == -1 || tm->tm_hour == st.hour) &&
-        (st.mday == -1 || tm->tm_mday == st.mday) && (st.month == -1 || (tm->tm_mon + 1) == st.month) &&
-        (st.wday == -1 || tm->tm_wday == st.wday)) {
+    bool minuteMatch = (st.minute == -1 || tm->tm_min == st.minute ||
+                        (st.minute > 0 && st.minute <= 59 && tm->tm_min % st.minute == 0) ||
+                        (st.minute == 0 && tm->tm_min == 0));
```

Comment:
`st.minute == 0 && tm->tm_min == 0` seems useless since it's covered in `tm->tm_min == st.minute`.

`st.minute > 0 && st.minute <= 59 && tm->tm_min % st.minute == 0`: from my understanding, I think we need to check `st.minute_interval` to see if it's `true`?